src/main/scala/io/iohk/ethereum/blockchain/sync/fast/FastSync.scala

```diff
@@ -359,13 +366,13 @@ class FastSync(
     }
 
     private def waitingForBranchResolution: Receive = handleStatus orElse {
-      case FastSyncBranchResolverActor.BranchResolvedSuccessful(firstCommonBlockNumber, resolvedPeer) =>
+      case FastSyncBranchResolverActor.BranchResolvedSuccessful(firstCommonBlockNumber, newMasterPeer) =>
         // Reset the batch failures count
         batchFailuresCount = 0
 
         // Restart syncing from the valid block available in state.
         syncState = syncState.copy(bestBlockHeaderNumber = firstCommonBlockNumber)
```

Comment:
I think we need to update `syncState.nextBlockToFullyValidate` as well here. 
maybe `nextBlockToFullyValidate = firstCommonBlockNumber + 1`

Reply:
yes, makes sense to update `nextBlockToFullyValidate` as well :+1: